python/cudf/cudf/core/frame.py

```diff
@@ -846,7 +868,7 @@ def _copy_categories(self, other, include_index=True):
             ):
                 self._data[name] = build_categorical_column(
                     categories=other_col.categories,
-                    codes=col,
+                    codes=as_column(col.base_data, dtype=col.dtype),
```

Comment:
This feels super awkward. Should we update the `build_categorical_column` API to better capture using base allocations instead of offsetted allocations?

Reply:
Or is this hiding the fact that something isn't additively applying the offsets within `CategoricalColumn`?

Reply:
@kkraus14 Made the `build_categorical_column` refactor changes in a separate PR: https://github.com/rapidsai/cudf/pull/4674